### PR TITLE
Registry dataset list on Network page

### LINKED ISSUE #421: lawl we totally forgot to put dataset renaming back in
So while we're here, might as well make renaming work properly.

I think we should add it as a text-link next to `remove` in the dataset action buttons, the link text should be in blue, not in red.

* It should present a modal that says "choose a new dataset name"
* the modal should display the current dataset name
* the modal should show validation errors when the name is invalid.
* the modal should have a cancel "x" (or a cancel button, whatever's easier) 
* The save button should flip to "loading" when the user hits submit, and the modal should dismiss on successful change
* the user should see a message that says "dataset successfully renamed"
* the user should be redirected back to the collections page

### LINKED ISSUE #425: add `datasets` column to `network` tab
This column should pull from the newly made `/registry/list` api endpoint, which returns a list of the dataset summaries available on the registry.

* [x] create new `actions/registry.js` file
* [x] add `registry` section to `entities`
* [x] add action to pagination
* [x] create action that calls to `/registry/list`
```
REGISTRY_DATASETS_REQUEST
REGISTRY_DATASETS_SUCCESS
REGISTRY_DATASETS_FAILURE
```
* [x] create new `selectors/registry.js` file
    * [ ] selectNetworkDatasets func => selects ids from pagination and gets datasets from entities
* [x] select datasets in network container
* [x] add datasets list component to network component

### PR DESCRIPTION
**Pull request by ramfox**
closes #425 
closes #421 

  * [x] sometimes there is weird behavior on the dataset history page when the dataset was not pulled from the network. Going to remove the history tab when the dataset was pulled from the registry
  * [x] Create function in qri that gets registry via endpoint 
  * [x] Create actions in frontend that gets registry dataset: 
  * [x] loadRegistryDatasetByName 
  * [x] loadRegistryDatasetByPath 
  * [x] make `fromRegistry` adjustments in DatasetHeader  

__________
edit: after talking with Brendan, going to remove the new registry section of entities, and have all the datasets live in one place. So much nicer.

Here are the other things that this pr covers:

  * [x] check for profile photo bug

  Walking back some network additions:
  * [x] remove registry schemas
  * [x] remove registry section from entities
  * [x] update selectors to pull from the correct section
  * [x] double check actions for any needed changes
  * [x] remove /network/ route
  * [x] remove RegistryDatasetContainer & any references
  * [x] remove anything that links to /network/ route 

  * [x] add isRegistryDataset(path) selector
      * should find if it is in the list of registry ids
      * if not, return false
      * if yes
        iterate through other dataset sections to see if it is also in there
      * if yes, return false
      * otherwise return true

   * [x] add isLocalDataset(path) selector
      * isLocalDataset should check the local pagination ids and if the hash exists in this location, then it is local
  * [x] isLocalDataset should check the local pagination ids and if the hash exists in this location, then it is local
      * if there is no hash => we need to check via name => for each hash, check select the dataset and check if the name and dataset name match. Return true if so.

   * [x] add isLocal var to dataset and dataset container 
   * [x] add fromRegistry var to dataset and dataset container 

   * [x] add datasetLinks function that determines the links to pass to the navLinks page 
    (using datasetRef, fromRegistry, and isLocal)
       * overview = always
       * body = filter out if !fromRegistry
       * meta => filter out if section not present
       * structure => filter out if section not present
       * transform => filter out if section not present or scriptPath not present in section
       * viz => filter out if section not present or scriptPath not present
       * history => filter out if !fromRegistry

   * [x] add buttonFunctions function that determines with functions to pass to the buttonGroup 
      (using isLocal, fromRegistry, isPublished, and isLatestDataset)
      * edit => isLatestDataset && !fromRegistry
      * add => !isLocal
      * delete => isLocal && isLatestDataset
      * publish => isLocal && inMyNameSpace && !isPublished
      * export => isLocal
      * rename => isLocal && isLatestDataset

### CODE REVIEW

File: lib/actions/registry.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    [CALL_API]: {
+      types: [REGISTRY_DATASET_REQUEST, REGISTRY_DATASET_SUCCESS, REGISTRY_DATASET_FAILURE],
+      endpoint: namedPath,
+      schema: Schemas.DATASET,
+      path
+    }
+  }
+}
+
+// TODO: look through registry and see if we already have this registry dataset before fetching
```
Comment (b5) [inline lib/actions/registry.js:57]: I haven't mentioned this yet, but @dustmop proposed carrying over a habit that I think we should take up:
switch from writing `TODO:` to `TODO (username):`. This way, when someone wants to address the TODO, they know who to ask for more details.

So no worries for this PR, but in the future I'd love to get in this habit

**Review by b5 — APPROVED** [review]
Ugh I'm sorry, I've got a bit too much going on in my neck of the woods to give this a proper review. How do you feel about this PR? I'd be happy to merge it and deal with the bugs in subsequent PR.

**Comment by ramfox** [comment]
I feel good about the pr in terms of the issues that is addresses. I've been dogfooding and seeing other bugs/issues, but out of the scope of what I was initially trying to address.

I say we merge and I'll start new prs for bugs that arise, especially since it will feel like forward momentum.

**Comment by b5** [comment]
perf. Let's do it!

**Pull request merged**